frontend/components/flashcard/Flashcard.tsx

```diff
@@ -1,155 +1,268 @@
 }
 
 /**
  * UI component for displaying flashcards
  */
-export const Card = ({
+export const Flashcard = ({
 	id,
 	card,
-	cardsleft,
+	dueCards,
 	cardSide = 0,
 	isEdit = false,
-	className,
-}: CardProps) => {
-	const [flashCard, setFlashCard] = useState(card);
-	const [tempCard, setTempCard] = useState(flashCard);
-	const [side, setSide] = useState(cardSide % tempCard.length);
-	const [edit, setEdit] = useState(isEdit);
+	fullSize = false,
+	className = "",
+	push,
+}: FlashcardProps) => {
+	const { mutate } = useSWRConfig();
 
-	function editField(field: "header" | "description", value: string) {
-		const card = structuredClone(tempCard);
-		card[side][field] = value;
-		setTempCard(card);
-	}
+	const [tempCard, setTempCard] = useState<Card>(card);
+	const [side, setSide] = useState<number>(cardSide % card.sides?.length);
+	const [edit, setEdit] = useState<boolean>(isEdit);
 
 	return (
 		<div
 			id={id}
-			className={`mx-auto my-auto flex h-5/6 w-5/6 flex-col space-y-3 rounded-xl border-2 border-kiokuDarkBlue bg-eggshell p-5 font-semibold text-kiokuDarkBlue shadow-lg sm:h-4/5 sm:w-4/5 md:h-3/4 md:w-3/4 lg:h-2/3 lg:w-2/3 lg:p-10 ${
-				className ?? ""
+			className={`mx-auto my-auto flex flex-col space-y-1 rounded-xl border-2 border-kiokuDarkBlue bg-eggshell p-3 font-semibold text-kiokuDarkBlue shadow-lg md:space-y-3 md:p-5 lg:p-10 ${className} ${
+				fullSize
+					? "h-full w-full"
+					: "h-5/6 w-5/6 sm:h-4/5 sm:w-4/5 md:h-3/4 md:w-3/4 lg:h-2/3 lg:w-2/3"
 			}`}
 		>
 			<div className="flex h-full flex-row justify-between">
 				<div className="flex w-full flex-col">
 					<div className="flex flex-row items-center justify-between">
-						<input
-							className="w-full border-0 bg-eggshell text-lg outline-none sm:text-xl md:text-2xl lg:text-3xl"
-							value={tempCard[side].header}
-							placeholder="Header"
+						<InputField
+							id="headerInputId"
+							type="text"
+							name="headerInput"
+							value={tempCard.sides[side]?.header}
+							placeholder={edit ? "Header" : ""}
+							style="secondary"
 							readOnly={!edit}
-							onChange={(event) =>
-								editField("header", event.target.value)
-							}
-						></input>
+							className="text-lg sm:text-xl md:text-2xl lg:text-3xl"
+							onChange={(event) => {
+								editField("header", event.target.value);
+							}}
+						></InputField>
 						{edit ? (
-							<div className="flex flex-row space-x-3">
-								<Check
-									className="hover:cursor-pointer"
-									onClick={() => {
-										setFlashCard(tempCard);
-										setEdit(false);
-									}}
-								></Check>
-								<X
+							<div className="flex flex-row items-center space-x-5">
+								{tempCard.sides.length > 1 && (
+									<FileMinus
+										id="deleteSideButtonId"
+										className="hover:cursor-pointer"
+										onClick={() => {
+											setSide(
+												Math.min(
+													side,
+													tempCard.sides.length - 2
+												)
```

Comment:
this can cause unexpected behaviour. rather do this:

```suggestion
											setSide(oldSide => Math.min(
													oldSide,
													tempCard.sides.length - 2
											)
```

Reply:
fixed in 7b69932b913ca6cf11db44fd1071d49e1ef06ade